hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -366,36 +382,33 @@ private void applyOrdering(TableGroup tableGroup, PluralAttributeMapping pluralA
 		}
 	}
 
-	private void applyOrdering(
-			TableGroup tableGroup,
-			OrderByFragment orderByFragment) {
+	private void applyOrdering(TableGroup tableGroup, OrderByFragment orderByFragment) {
 		if ( orderByFragments == null ) {
 			orderByFragments = new LinkedHashMap<>();
 		}
 		orderByFragments.put( orderByFragment, tableGroup );
 	}
 
-	private int fetchDepth = 0;
-
 	private List<Fetch> visitFetches(FetchParent fetchParent, QuerySpec querySpec, LoaderSqlAstCreationState creationState) {
 		log.tracef( "Starting visitation of FetchParent's Fetchables : %s", fetchParent.getNavigablePath() );
 
 		final List<Fetch> fetches = new ArrayList<>();
 
-		final Consumer<Fetchable> processor = createFetchableConsumer( fetchParent, querySpec, creationState, fetches );
+		final BiConsumer<Fetchable, Boolean> processor = createFetchableBiConsumer( fetchParent, querySpec, creationState, fetches );
 
 		final FetchableContainer referencedMappingContainer = fetchParent.getReferencedMappingContainer();
-		referencedMappingContainer.visitKeyFetchables( processor, null );
-		referencedMappingContainer.visitFetchables( processor, null );
+		referencedMappingContainer.visitKeyFetchables( fetchable -> processor.accept( fetchable, true ), null );
+		referencedMappingContainer.visitFetchables( fetchable -> processor.accept( fetchable, false ), null );
```

Comment:
The boolean parameter to denote whether we are exploring 'keySubGraph' is important to disambiguate in entity graph implementing, so I changed `Consumer` to `BiConsumer`.